xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h

```diff
@@ -1547,15 +1548,45 @@ namespace vfs
         kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before GetProperty!");
         return "";
       }
-      char *res(::kodi::addon::CAddonBase::m_interface->toKodi->kodi_filesystem->get_property(
-        ::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, m_file, type, name.c_str()));
+      std::vector<std::string> values = GetPropertyValues(type, name);
+      if (values.empty()) {
+        return "";
+      }
+      return values[0];
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve file property values
+    ///
+    /// @param[in] type         The type of the file property values to retrieve the value for
+    /// @param[in] name         The name of the named property (e.g. Header)
+    /// @return                 values of requested property, empty vector on failure / non-existance
+    ///
+    const std::vector<std::string> GetPropertyValues(FilePropertyTypes type, const std::string &name) const
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before GetPropertyValues!");
+        return std::vector<std::string>();
+      }
+      int numValues;
+      char **res(::kodi::addon::CAddonBase::m_interface->toKodi->kodi_filesystem->get_property_values(
+        ::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, m_file, type, name.c_str(), &numValues));
       if (res)
       {
-        std::string strReturn(res);
-        ::kodi::addon::CAddonBase::m_interface->toKodi->free_string(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, res);
-        return strReturn;
+        std::vector<std::string> vecReturn;
+        for (int i = 0; i < numValues; i++)
+        {
+          vecReturn.push_back(res[i]);
+          ::kodi::addon::CAddonBase::m_interface->toKodi->free_string(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, res[i]);
+        }
+        ::kodi::addon::CAddonBase::m_interface->toKodi->free_string(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, (char*)res);
```

Comment:
Is it ok the way I free the string-array here? The contained elements are already freed above.

Reply:
Not sure. `free_string` is, as the name implies, for freeing strings (char*), not arbitrary heap allocated objects. @AlwinEsch ?

Reply:
Question is if freestring is capable of freeing : char **ret = new char*[*numValues];
I don't believe so, but just check freestring impl. and you know, but I think you have to use an malloc instead new[]

Reply:
```
void CAddonDll::free_string(void* kodiBase, char* str)
{
  if (str)
    free(str);
}
```
I would say its technically ok.

Reply:
No it's not. new[] and free do not fit together!

Reply:
Oh, i see, free may only be used with malloc... Changed the allocation to malloc.

Reply:
array vs non-array, c++ vs libc.

Reply:
`malloc`is technically okay, then. 

But my clear -1 for using `free_string`to free the array - for **semantical** reasons. `free_string`'s semantics is to free a single string not an array and the fact that it internally uses `free` which does technically "fit" is nothing more than an implementation detail that can change anytime.

Sorry, but imo we need another API function`free_stringarray`, which does the whole job, freeing the array elements and the array itself.